app/serializers/disability_claim_detail_serializer.rb

```diff
@@ -51,30 +51,44 @@ def create_event_from_string_date(type, *from_keys)
   ).freeze
 
   def create_events_for_tracked_items
-    events = []
-    TRACKED_ITEM_FIELDS.each do |field|
-      list_objects_with_key(['claimTrackedItems', field], ['openedDate']).each do |obj|
-        date = obj['openedDate'] || obj['receivedDate']
-        events << {
-          type: field.snakecase,
-          date: Date.strptime(date, '%m/%d/%Y'),
-          description: obj['description'],
-          display_name: obj['displayedName'],
-          overdue: obj['overdue'],
-          tracked_item_id: obj['trackedItemId'],
-          tracked_item_status: obj['trackedItemStatus']
-        } if date
+    TRACKED_ITEM_FIELDS.map do |field|
+      sub_objects_of('claimTrackedItems', field).map do |obj|
+        create_tracked_item_event(field.snakecase, obj)
       end
-    end
-    events
+    end.flatten
+  end
+
+  def create_tracked_item_event(type, obj)
+    event = {
+      type: type,
+      tracked_item_id: obj['trackedItemId'],
+      description: obj['description'],
+      display_name: obj['displayedName'],
+      overdue: obj['overdue'],
+      status: obj['trackedItemStatus'],
+      uploaded: obj['uploaded'],
+      uploads_allowed: obj['uploadsAllowed'],
+      opened_date: date_or_nil_from(obj, 'openedDate'),
+      requested_date: date_or_nil_from(obj, 'requestedDate'),
+      received_date: date_or_nil_from(obj, 'receivedDate'),
+      closed_date: date_or_nil_from(obj, 'closedDate'),
+      suspense_date: date_or_nil_from(obj, 'suspenseDate')
+    }
+    event[:date] = [
+      event[:opened_date], event[:requested_date], event[:received_date],
```

Comment:
Does the order here matter or are you just looking for any date that is present? I'm not sure how complicated we want to make this, it could also just be a series of `||` or slice the event hash to be a bit more concise here, or just leave it as you have it 
```ruby
EVENT_DATE_FIELDS = %i(opened_date requested_date ...)

event[:date] = event.slice(*EVENT_DATE_FIELDS).values.compact.first
```